maps/_common/mapsystem/map.dm

```diff
@@ -166,6 +166,7 @@
 		var/exoplanet_type = pick(possible_exoplanets)
 		log_debug("Building new exoplanet with type: [exoplanet_type] and size: [planet_size[1]] [planet_size[2]]")
 		var/obj/effect/overmap/visitable/sector/exoplanet/new_planet = new exoplanet_type(null, planet_size[1], planet_size[2])
+		possible_exoplanets -= new_planet
```

Comment:
Are you sure that will remove it from the list? `possible_exoplanets` is a type based list, `new_planet` is a reference to an object of the type. I think you need to use `new_planet.type`

Reply:
It runtimes now because the list is empty somehow.